src/compile/axis/properties.ts

```diff
@@ -167,58 +169,111 @@ export function getLabelAngle(
   }
 }
 
+export function normalizeAngleExpr(angle: SignalRef) {
+  return `(((${angle.signal} % 360) + 360) % 360)`;
+}
+
 export function defaultLabelBaseline(
-  angle: number,
-  axisOrient: AxisOrient,
+  angle: number | SignalRef,
+  orient: AxisOrient | SignalRef,
   channel?: 'x' | 'y',
   alwaysIncludeMiddle?: boolean
 ) {
-  channel = channel || (axisOrient === 'top' || axisOrient === 'bottom' ? 'x' : 'y');
+  channel = channel || (orient === 'top' || orient === 'bottom' ? 'x' : 'y');
 
   if (angle !== undefined) {
     if (channel === 'x') {
-      return (45 < angle && angle < 135) || (225 < angle && angle < 315)
-        ? 'middle'
-        : (angle <= 45 || 315 <= angle) === (axisOrient === 'top')
-        ? 'bottom'
-        : 'top';
+      if (isSignalRef(angle)) {
+        const a = normalizeAngleExpr(angle);
+        const orientIsTop = isSignalRef(orient) ? `(${orient.signal} === "top")` : orient === 'top';
+        return {
+          signal:
+            `(45 < ${a} && ${a} < 135) || (225 < ${a} && ${a} < 315) ? "middle" :` +
+            `(${a} <= 45 || 315 <= ${a}) === ${orientIsTop} ? "bottom" : "top"`
```

Comment:
Parenthesize here too.

Reply:
(And etc. for other occurrences.)

Reply:
Can you come up with a case that will break this?
Otherwise, I'd rather have the output readable.

Reply:
Actually ternary operator will break this.

Reply:
Let's deal with this in #6374